Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -297,11 +303,14 @@ extension PackageDependencyDescription {
         }
 
         let requirement = try Requirement(v4: json.get("requirement"))
+        let url = try fixURL(json.get("url"), requirement: requirement)
+        var name: String? = json.get("name")
+
+        if name == nil && toolsVersion >= .v5_3 {
+            name = URL(string: url)?.deletingPathExtension().lastPathComponent
```

Comment:
Maybe make a static method on PackageReference, similar to `computeIdentity(packageURL:)`